src/components/common/ErrorBoundary/index.tsx

```diff
@@ -37,8 +41,18 @@ const ErrorBoundary: FallbackRender = ({ error, componentStack }) => {
           </>
         )}
         <Typography mt={2}>
-          <Link href={AppRoutes.welcome} passHref target="_blank" rel="noopener noreferrer" color="primary">
-            <MuiLink>Go Home</MuiLink>
+          <Link href={AppRoutes.welcome} passHref color="primary">
+            <MuiLink
+              onClick={(e) => {
+                e.stopPropagation()
+
+                // We need to wait for navigation to finish otherwise error will be thrown
+                // from the current page again
+                router.push(AppRoutes.welcome).then(resetError)
```

Comment:
Why does it have to be a JS function and not just a link?
I would even make it an actual link, that does a full page reload.

Reply:
Resetting the error occurs before the navigation and the error is thrown again from what was the previous page.

If you'd sooner use a link then this is unnecessary then. I'll change it.